docs/decisions/0007-sys-path-modification-removal.rst

```diff
@@ -32,11 +34,190 @@ This deprecation will take place in the following steps:
 
 4. Fix all instances where the ``sys.path``-based modules were ``patch``-ed in unit tests, as those patches no longer work.
 
-5. Once all instances of the ``sys.path``-based imports have been migrated in the ``edx-platform`` codebase, officially deprecate that import pattern for the next open-source release (as per the standard deprecation process).
+5. Once all instances of the ``sys.path``-based imports have been migrated in the ``edx-platform`` codebase, officially deprecate that import pattern for the next open-source release.
 
 6. Monitor logs to address continued use of the ``sys.path`` based import patterns.
 
 Goals
 =====
 
 Eliminate modifications to ``sys.path`` to enable easier tool use in edx-platform.
+
+
+Timeline
+========
+
+In the Koa release, usages of the old import style raised instances of
+``DeprecatedEdxPlatformImportWarning``, but still worked.
+
+In the Lilac release, usages old import paths will raise instances of
+``DeprecatedEdxPlatformImportError``, breaking any code that was not updated.
+This error class is intentionally *not* a subclass of ``ImportError``;
+to understand why, consider the following common pattern used in packages outside
+of the edx-platform repo itself::
+  try:
+      from student.models import CourseEnrollment
+  except ImportError:
+      CourseEnrollment = None
+This pattern is (unfortunately) widely used to so packages can import
+objects from edx-platform modules in developer or production environments,
+whilst falling back to ``None`` when unit tests are run.
+Now, if the the above old-style import statement raised a subclass of
+``ImportError``, then it would be silently caught by the ``except`` clause!
+By instead raising from a disjoint error class in the Lilac release,
+we make it more likely that operators can quickly notice and fix
+any lingering old-style import paths as they migrate from Koa.
+
+In Maple and beyond, old import paths will simply raise ``ImportError``,
+allowing us to remove the ``import_shims`` directory tree.
+
+
+Upgrade Guide
+=============
+
+Where to look for old import paths:
+
+* Forks of ``edx-platform`` itself.
+* Packages that import edx-platform code (``edx-completion``, ``edx-enterprise``, et al).
```

Comment:
I recently broke Enterprise stuff, I assume you've made sure that team is aware of these changes?

Reply:
Yes :+1: I've had a few "you broke enterprise!" moments myself haha. I've scoured the edx GitHub org and Splunk for usages of the old imports over the past few months.